frdrpc/macaroons.go

```diff
@@ -0,0 +1,160 @@
+package frdrpc
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"google.golang.org/grpc"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+)
+
+const (
+	// faradayMacaroonLocation is the value we use for the faraday
+	// macaroons' "Location" field when baking them.
+	faradayMacaroonLocation = "faraday"
+)
+
+var (
+	// RequiredPermissions is a map of all faraday RPC methods and their
+	// required macaroon permissions to access faraday.
+	RequiredPermissions = map[string][]bakery.Op{
+		"/frdrpc.FaradayServer/OutlierRecommendations": {{
+			Entity: "recommendation",
+			Action: "read",
+		}},
+		"/frdrpc.FaradayServer/ThresholdRecommendations": {{
+			Entity: "recommendation",
+			Action: "read",
+		}},
+		"/frdrpc.FaradayServer/RevenueReport": {{
+			Entity: "report",
+			Action: "read",
+		}},
+		"/frdrpc.FaradayServer/ChannelInsights": {{
+			Entity: "insights",
+			Action: "read",
+		}},
+		"/frdrpc.FaradayServer/ExchangeRate": {{
+			Entity: "rates",
+			Action: "read",
+		}},
+		"/frdrpc.FaradayServer/NodeAudit": {{
+			Entity: "audit",
+			Action: "read",
+		}},
+		"/frdrpc.FaradayServer/CloseReport": {{
+			Entity: "report",
+			Action: "read",
+		}},
+	}
+
+	// allPermissions is the list of all existing permissions that exist
+	// for faraday's RPC. The default macaroon that is created on startup
+	// contains all these permissions and is therefore equivalent to lnd's
+	// admin.macaroon but for faraday.
+	allPermissions = []bakery.Op{{
+		Entity: "recommendation",
+		Action: "read",
+	}, {
+		Entity: "report",
+		Action: "read",
+	}, {
+		Entity: "audit",
+		Action: "read",
+	}, {
+		Entity: "insights",
+		Action: "read",
+	}, {
+		Entity: "rates",
+		Action: "read",
+	}}
+
+	// macDbDefaultPw is the default encryption password used to encrypt the
+	// faraday macaroon database. The macaroon service requires us to set a
+	// non-nil password so we set it to an empty string. This will cause the
+	// keys to be encrypted on disk but won't provide any security at all as
+	// the password is known to anyone.
+	//
+	// TODO(guggero): Allow the password to be specified by the user. Needs
+	// create/unlock calls in the RPC. Using a password should be optional
+	// though.
+	macDbDefaultPw = []byte("")
+)
+
+// startMacaroonService starts the macaroon validation service, creates or
+// unlocks the macaroon database and creates the default macaroon if it doesn't
+// exist yet. If macaroons are disabled in general in the configuration, none of
+// these actions are taken.
+func (s *RPCServer) startMacaroonService() error {
```

Comment:
is "if macaroons are disabled.." relevant here since we don't have a `--no-macaroons` option?

Reply:
Right, that was left over from when we still had the `--no-macaroons` in the loop PR.